app/controllers/dashboards_controller.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+# Controller for Dashboards
+class DashboardsController < ApplicationController
+  before_action :authorize!
+
+  def show
+    admin_metrics unless current_user.student?
+    student_variables unless current_user.admin?
+  end
+
+  private
+
+  def admin_metrics
+    @draws = Draw.all.includes(:groups).sort_by(&:name)
+    @draw_hash = Hash.new { |hash, key| hash[key] = {} }
+    @draws.each do |d|
+      @draw_hash[d][:partial] = d.status
+      if %w(pre_lottery suite_selection results).include? d.status
+        send("#{d.status}_metrics".to_sym, d, @draw_hash[d])
+      end
+    end
+  end
+
+  # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
```

Comment:
If you want to put this inline you can get rid of the `Metrics/` prefix for both cops.